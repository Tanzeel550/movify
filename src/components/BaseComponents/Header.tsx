import React from 'react';
import { NavLink } from 'react-router-dom';

import { startLogout } from '../../actions/authActions';
import { connect } from 'react-redux';
import VerifyEmailBtn from '../Utils/VerifyEmailBtn';

type Props = {
  isAuthenticated: boolean;
  emailVerified: boolean;
};

const Img =
  'https://drive.google.com/file/d/1D-SBsPY5UCyPl-fSo7NHa4c8szsZI3Ig/view?usp=sharing';

const Header = ({ isAuthenticated, emailVerified }: Props) => (
  <header className="header">
    <div className="container header__container">
      <div className="header--left">
        <NavLink to="/" className="header--left__item">
          <img src={Img} alt="" className="header--left__item-img" />
          <h2 className="header--left__item-text">Movify</h2>
        </NavLink>
      </div>

      {isAuthenticated ? (
        <div className="header--right">
          {!emailVerified && <VerifyEmailBtn />}
          <NavLink to="/create" className="u-btn-colorized">
            Create
          </NavLink>
          <button
            className="u-btn-colorized u-btn-colorized-active"
            onClick={startLogout}
          >
            Logout
          </button>
        </div>
      ) : (
        <div className="header--right">
          <NavLink to="/signUp" className="u-btn-colorized">
            SignUp
          </NavLink>
          <NavLink
            to="login"
            className="u-btn-colorized u-btn-colorized-active"
          >
            Login
          </NavLink>
        </div>
      )}
    </div>
  </header>
);

type stateType = {
  auth: {
    isAuthenticated: boolean;
    user: {
      emailVerified: boolean;
    };
  };
};

const mapStateToProps = ({ auth }: stateType) => ({
  isAuthenticated: auth.isAuthenticated,
  emailVerified: auth.user.emailVerified,
});

export default connect(mapStateToProps)(Header);
