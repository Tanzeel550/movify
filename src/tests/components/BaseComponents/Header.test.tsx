import { shallow } from 'enzyme';
import { Header } from '../../../components/BaseComponents/Header';
import React from 'react';

test('should render header correctly when user is logged in and email is verified', () => {
  expect(
    shallow(<Header isAuthenticated={true} emailVerified={true} />)
  ).toMatchSnapshot();
});

test('should render header correctly when user is logged in but email is not verified', () => {
  expect(
    shallow(<Header isAuthenticated={true} emailVerified={false} />)
  ).toMatchSnapshot();
});

test('should render header correctly when user is logged out', () => {
  expect(
    shallow(<Header isAuthenticated={false} emailVerified={false} />)
  ).toMatchSnapshot();
});
