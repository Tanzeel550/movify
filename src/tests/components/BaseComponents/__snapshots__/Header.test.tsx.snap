// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly when user is logged in and email is verified 1`] = `
<header
  className="header"
>
  <div
    className="container header__container"
  >
    <div
      className="header--left"
    >
      <NavLink
        className="header--left__item"
        to="/"
      >
        <img
          alt=""
          className="header--left__item-img"
          src="icon.jpg"
        />
        <h2
          className="header--left__item-text"
        >
          Movify
        </h2>
      </NavLink>
    </div>
    <div
      className="header--right"
    >
      <NavLink
        className="u-btn-colorized"
        to="/create"
      >
        Create
      </NavLink>
      <button
        className="u-btn-colorized u-btn-colorized-active"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;

exports[`should render header correctly when user is logged in but email is not verified 1`] = `
<header
  className="header"
>
  <div
    className="container header__container"
  >
    <div
      className="header--left"
    >
      <NavLink
        className="header--left__item"
        to="/"
      >
        <img
          alt=""
          className="header--left__item-img"
          src="icon.jpg"
        />
        <h2
          className="header--left__item-text"
        >
          Movify
        </h2>
      </NavLink>
    </div>
    <div
      className="header--right"
    >
      <Connect(VerifyEmailBtn) />
      <NavLink
        className="u-btn-colorized"
        to="/create"
      >
        Create
      </NavLink>
      <button
        className="u-btn-colorized u-btn-colorized-active"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;

exports[`should render header correctly when user is logged out 1`] = `
<header
  className="header"
>
  <div
    className="container header__container"
  >
    <div
      className="header--left"
    >
      <NavLink
        className="header--left__item"
        to="/"
      >
        <img
          alt=""
          className="header--left__item-img"
          src="icon.jpg"
        />
        <h2
          className="header--left__item-text"
        >
          Movify
        </h2>
      </NavLink>
    </div>
    <div
      className="header--right"
    >
      <NavLink
        className="u-btn-colorized"
        to="/signUp"
      >
        SignUp
      </NavLink>
      <NavLink
        className="u-btn-colorized u-btn-colorized-active"
        to="login"
      >
        Login
      </NavLink>
    </div>
  </div>
</header>
`;
